01_AppCore/Business/Models/Results/ErrorResult.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+namespace _01_AppCore.Business.Models.Results
+{
+    public class ErrorResult : Result
+    {
+        public ErrorResult(string message) : base(ResultStatus.Error, message)
```

Comment:
I don't think there's a need to have 2 entirely separate classes (ErrorResult and SuccessResult) that are actually the same but with a different enum value.
Keep a single result class and provider 2 static factory methods on it ( Result.Ok() and Result.Error()) and build the Result object in them with a correct enum value